pkg/framework/host.go

```diff
@@ -177,17 +178,76 @@ func (h *Host) CreateNamespace(ns string) error {
 	return nil
 }
 
-func (h *Host) DeleteGuestCluster(name, cr, logEntry string) error {
-	if err := runCmd(fmt.Sprintf("kubectl delete %s %s", cr, h.clusterID)); err != nil {
-		return microerror.Mask(err)
+func (h *Host) DeleteGuestCluster(ctx context.Context, provider string) error {
```

Comment:
I'm thinking about making `provider` a mandatory `Host` field. Is it a problem for any e2e test we run?

Reply:
Kind of, we have e2e tests that don't need a guest cluster (e.g. operatorkit or chart-operator)